### ISSUE TITLE
reverse_delete_rule fails on traversing references

### ISSUE THREAD

**Issue by wpjunior** [hmarr/mongoengine#227]
not fails:

``` python
from mongoengine import *
connect('teste')

class Parents(EmbeddedDocument):
    pai_ref = ReferenceField('Person')

    mae_ref = ReferenceField('Person')

    meta = {'allow_inheritance': False}

class Person(Document):
    parents = EmbeddedDocumentField('Parents')

if __name__ == '__main__':
    p1 = Person()
    p1.save()

    p2 = Person(parents=Parents(pai_ref=p1))
    p2.save()

```

fails:

``` python
from mongoengine import *
connect('teste')

class Parents(EmbeddedDocument):
    pai_ref = ReferenceField('Person',
                             db_field="paiRef",
                             reverse_delete_rule=NULLIFY)

    mae_ref = ReferenceField('Person',
                             db_field="maeRef",
                             reverse_delete_rule=NULLIFY)

    meta = {'allow_inheritance': False}

class Person(Document):
    parents = EmbeddedDocumentField('Parents')

if __name__ == '__main__':
    p1 = Person()
    p1.save()

    p2 = Person(parents=Parents(pai_ref=p1))
    p2.save()

```

**Comment by rozza** [comment]
Its due to `reverse_delete_rule` needing the class of the Reference field to attach the delete rule to it...

**Comment by rozza** [comment]
@wpjunior ha tricky one this one, I initially fixed by adding a custom **isinstance** as calling a classmethod in **new** seems to break isinstance (even if the classmethod does nothing!).

Only then did I find we dont support reverse_delete_rules for embedded documents as we can't finding the class the embedded documents is in and the field name(s) where it is referenced..

So now we raise an InvalidDocumentError warning you this is the case...

Sorry, marking as a won't fix for 0.6
